### ISSUE TITLE
MIDI freezes apps on exit

### ISSUE THREAD

**Issue by IgorX2** [MicrosoftDocs/windows-uwp#1424]
MIDI does not work properly and freezes apps on exit. When using a minimal example with MIDI (manually assign an input device, register message handling and process input), and exiting the app, the app freezes and cannot be terminated from command line or Task Manager (access denied, while the process only has 1 thread and 2 handles). Further, the app cannot be redeployed from Visual Studio, as the entire MIDI stack seems to freeze. Computer restart must be done before the app can be deployed again. This happens in the most basic of examples. Also, Note Off events are not properly received.

This means that an app with basic MIDI input cannot be implemented, making the UWP platform completely useless for any music input, composition or production purposes.

The post on MSDN cannot be implemented: [Create MIDI ports to send and receive messages
](https://docs.microsoft.com/en-us/windows/uwp/audio-video-camera/midi)

**Comment by oldnewthing** [comment]
I cannot reproduce this problem. See [UWP Samples issue 1056](https://github.com/Microsoft/Windows-universal-samples/issues/1056).

**Comment by IgorX2** [comment]
@oldnewthing You need to connect an external MIDI device (like a MIDI keyboard). It won't hang with the virtual instrument.

**Comment by IgorX2** [comment]
@oldnewthing You need to connect a physical MIDI device (like a MIDI keyboard) to reproduce this.

**Comment by drewbatgit** [comment]
Thanks for taking the time to let us know about this and we sincerely apologize for the delayed response. Due to the sheer volume of feedback in our repo, we are currently prioritizing new Issues. If you believe your feedback is still actionable, let us know either directly in this thread or by filing a new Issue and we’ll work to incorporate the suggestions into the documentation. Though the Issue will be marked as Closed, I will monitor any replies that come afterwards so feel free to treat this like an open Issue.

Hello, GitHub issues in this repository are for feedback specifically about the documentation. It seems like your issue is a developer support request. Here are all of the developer support resources:

- The [Microsoft Q&A](https://docs.microsoft.com/en-us/answers/products/) site is the home for technical questions and answers at Microsoft.
- The [Feedback Hub app](https://support.microsoft.com/help/4021566/windows-10-send-feedback-to-microsoft-with-feedback-hub-app) is the right place to lobby for a change, a feature, or to make a suggestion.
- For community assistance with development questions, there are our [forums](https://social.msdn.microsoft.com/Forums/newthread?category=windowsapps&forum=wpdevelop&prof=required), and [StackOverflow](https://stackoverflow.com/questions).
- The [Windows developer support](https://developer.microsoft.com/windows/support) site explains other support options.
- The [Visual Studio developer community](https://developercommunity.visualstudio.com) is available, and responsive. The former User Voice feature is now there on the **Suggest a feature** tab, if you want to lobby for a change, a feature, or to make a suggestion.
- For developer issues, or suspected product bugs, there's [Microsoft Support](https://support.microsoft.com/contactus); a support engineer will look at the repro and see whether the issue is a Microsoft product bug, or something else. If it's a bug with a Microsoft product then the ticket will be free of charge.

If you receive assistance via any support resource, and you believe that information would help us improve the documentation, then please come back and let us know.
